twins/service.go

```diff
@@ -326,20 +330,29 @@ func prepareState(st *State, tw *Twin, rec senml.Record, msg *mainflux.Message)
 		}
 	}
 
-	save := false
+	recTime := (rec.BaseTime + rec.Time) * 1000
+	action := 0 // 0 - do nothing, 1 - update, 2 - save
```

Comment:
You could add this action in the State to don't have to manipulate it as a separate variable

Reply:
This is the action ON the state and not OF the state. State does not know whether it needs to be updated or saved and does not need to keep this information.